File: CabService/src/main/java/co/inteleap/course/Statement.java
package co.inteleap.course;

import java.util.List;
import java.util.function.Function;
import java.util.function.Supplier;
import java.util.function.ToIntFunction;
import java.util.stream.Collectors;

public class Statement {

    List<Ride> rides;

    public Statement(List rides) {
        this.rides = rides;
    }

    public double getTotalFare()
    {
        double totalfare=0;
        for(Ride ride:rides)
            totalfare+= ride.getTotalFare.get();

        return totalfare;
    }


    public Supplier<Integer> getNumberOfRides =
            () -> rides.size();

    public Supplier<Double> getAvgFarePerRide =
            () -> getTotalFare() / getNumberOfRides.get();


}

File: TDD-Kotlin/src/main/kotlin/Statement.kt
data class Statement(val rides:List<Ride>) {

    fun totalFare(): Double {
        return rides.map(Ride::Fare).sum()
    }

    fun getNoOfRides(): Int {
        return rides.size
    }

    fun getAvgFarePerRide():Double{
        return totalFare()/getNoOfRides()
    }

}